libazureinit/src/provision/user.rs

```diff
@@ -96,26 +123,84 @@ impl Provisioner {
     }
 }
 
+#[instrument(skip_all)]
+fn user_exists(username: &str) -> Result<bool, Error> {
+    let status = Command::new("getent")
+        .arg("passwd")
+        .arg(username)
+        .status()?;
+
+    Ok(status.success())
+}
+
 #[instrument(skip_all)]
 fn useradd(user: &User) -> Result<(), Error> {
+    if user_exists(&user.name)? {
+        tracing::info!(
+            "User '{}' already exists. Skipping user creation.",
+            user.name
+        );
+
+        let group_list = user.groups.join(",");
+
+        tracing::info!(
+            "User '{}' is being added to the following groups: {}",
+            user.name,
+            group_list
+        );
+
+        let usermod_command =
+            format!("usermod -aG {} {}", group_list, user.name);
+
+        tracing::debug!("Running command: {}", usermod_command);
+
+        let status = Command::new("usermod")
+            .arg("-aG")
+            .arg(&group_list)
+            .arg(&user.name)
+            .status()?;
+
+        tracing::debug!("usermod command exit status: {}", status);
+
+        if !status.success() {
+            return Err(Error::SubprocessFailed {
+                command: usermod_command,
+                status,
+            });
+        }
+
+        return Ok(());
+    }
+
     let path_useradd = env!("PATH_USERADD");
     let home_path = format!("/home/{}", user.name);
 
+    let useradd_command = format!(
+        "{} {} --comment 'Provisioning agent created this user based on username provided in IMDS' --groups {} -d {} -m",
```

Comment:
NIT: for the comment, let's make it state `azure-init` created this user based on username provided by IMDS`